Material3

theimpulson opened the pull request:
## Sumary

This PR migrates Seedvault to use Material3.

## Issues

- https://github.com/seedvault-app/seedvault/issues/644

## Test

Everything passes locally

theimpulson committed:
Migrate to MaterialAlertDialog everywhere

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
themes: Switch to single theme for all activities

Default to NoActionBar theme and use toolbar where required

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
themes: Switch to Material3 theme

* Merge accent and primary color for simpler color management
* Apply background color on status and navigation bar
* Drop non-required colors

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
RecoveryCodeInputFragment: Switch to MaterialAutoCompleteTextView

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson commented:
Switches look really bad right now. At best I can copy the colors from datura and design from AOSP's settingsLib but unless we have gradle builds or AOSP updates the material components library, it is going to look bad.

theimpulson committed:
SchedulingFragment: Implement Material3 based ListPreference

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

grote commented on app/src/main/java/com/stevesoltys/seedvault/settings/SchedulingFragment.kt:
```diff
@@ -42,6 +45,21 @@ class SchedulingFragment : PreferenceFragmentCompat(),
+            else -> super.onDisplayPreferenceDialog(preference)
+        }
+    }
+
```
Google is really not giving us an easier way to get an MD3 list preference?

theimpulson replied:
Nope, none of the preferences are ready for MD3 stuff even in the latest release of the library. In Aurora Store, we have a similar class for EditTextPreference too for MD3 style.

grote commented:
```
> Task :app:ktlintMainSourceSetCheck FAILED
app/src/main/java/com/stevesoltys/seedvault/settings/preference/M3ListPreference.kt:47:1 Needless blank line(s) (no-consecutive-blank-lines)
```

grote committed:
Use MD3 switches in settings

grote committed:
Use MD3 switches for files backup config

grote commented:
colors seem wrong here:

![Screenshot_20240624_102718](https://github.com/seedvault-app/seedvault/assets/244947/d77ab834-f6cb-4d83-a5e4-5204e7ce3019)

grote commented:
For some reason, the exclamation mark error icon isn't red anymore: ![Screenshot_20240624_103537](https://github.com/seedvault-app/seedvault/assets/244947/aed14722-ae77-47c2-b77c-16ec795a7725)

In dark theme it is even worse, because dark greyish.

theimpulson committed:
drawable: Tint colors directly instead of relying on system theme

Material theme seems to be not respecting the tint color in the XML.
Override the colors manually to get the job done.

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson committed:
fragment_restore_app_selection: Set background color to squashed layout

CollapsingToolbarLayout doesn't seems to be respecting background color for some reason

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

theimpulson commented:
The only issue I can see right now is MD3 switches having wrong color in dark theme.

theimpulson committed:
Apply dynamic colors to all acitvites

Signed-off-by: Aayush Gupta <aayushgupta219@gmail.com>

grote reviewed (approved):
Awesome, thanks!
